frontend/src/types/cabins.ts

```diff
@@ -0,0 +1,18 @@
+import {
```

Comment:
Yeah, I went back and forth on this, but in the end I figured that if we're using generated types a lot anyway, I'd rather we name them in `types` rather than having `Fragment` and `ReturnOf<typeof ...Document>["property"]` scattered all around. In my opinion at least, the former leads to more readable and less cluttered code. However, I do see your point that having this central source for types may lead to bad GraphQL practice. I think it's a bigger refactor to move away from this, since a lot of our codebase is built around shared types, so I think that falls outside the scope of this PR.